packages/material-ui/src/Slider/SliderBase.js

```diff
@@ -0,0 +1,922 @@
+  return ((value - min) * 100) / (max - min);
+}
+
+function percentToValue(percent, min, max) {
+  return (max - min) * percent + min;
+}
+
+function getDecimalPrecision(num) {
+  // This handles the case when num is very small (0.00000001), js will turn this into 1e-8.
+  // When num is bigger than 1 or less than -1 it won't get converted to this notation so it's fine.
+  if (Math.abs(num) < 1) {
+    const parts = num.toExponential().split('e-');
+    const matissaDecimalPart = parts[0].split('.')[1];
+    return (matissaDecimalPart ? matissaDecimalPart.length : 0) + parseInt(parts[1], 10);
+  }
+
+  const decimalPart = num.toString().split('.')[1];
+  return decimalPart ? decimalPart.length : 0;
+}
+
+function roundValueToStep(value, step, min) {
+  const nearest = Math.round((value - min) / step) * step + min;
+  return Number(nearest.toFixed(getDecimalPrecision(step)));
+}
+
+function setValueIndex({ values, source, newValue, index }) {
+  // Performance shortcut
+  if (source[index] === newValue) {
+    return source;
+  }
+
+  const output = values.slice();
+  output[index] = newValue;
+  return output;
+}
+
+function focusThumb({ sliderRef, activeIndex, setActive }) {
+  const doc = ownerDocument(sliderRef.current);
+  if (
+    !sliderRef.current.contains(doc.activeElement) ||
+    Number(doc.activeElement.getAttribute('data-index')) !== activeIndex
+  ) {
+    sliderRef.current.querySelector(`[role="slider"][data-index="${activeIndex}"]`).focus();
+  }
+
+  if (setActive) {
+    setActive(activeIndex);
+  }
+}
+
+const axisProps = {
+  horizontal: {
+    offset: (percent) => ({ left: `${percent}%` }),
+    leap: (percent) => ({ width: `${percent}%` }),
+  },
+  'horizontal-reverse': {
+    offset: (percent) => ({ right: `${percent}%` }),
+    leap: (percent) => ({ width: `${percent}%` }),
+  },
+  vertical: {
+    offset: (percent) => ({ bottom: `${percent}%` }),
+    leap: (percent) => ({ height: `${percent}%` }),
+  },
+};
+
+const Identity = (x) => x;
+
+// TODO: remove support for Safari < 13.
+// https://caniuse.com/#search=touch-action
+//
+// Safari, on iOS, supports touch action since v13.
+// Over 80% of the iOS phones are compatible
+// in August 2020.
+let cachedSupportsTouchActionNone;
+function doesSupportTouchActionNone() {
+  if (cachedSupportsTouchActionNone === undefined) {
+    const element = document.createElement('div');
+    element.style.touchAction = 'none';
+    document.body.appendChild(element);
+    cachedSupportsTouchActionNone = window.getComputedStyle(element).touchAction === 'none';
+    element.parentElement.removeChild(element);
+  }
+  return cachedSupportsTouchActionNone;
+}
+
+const getUtilityClass = (name) => {
+  return `MuiSlider-${name}`;
+};
+
+const useSliderClasses = (props) => {
+  const { color, disabled, marked, orientation, track, classes } = props;
+
+  const overridesClasses = {
+    root: cx(
+      getUtilityClass('root'),
+      classes.root,
```

Comment:
I think that we can simplify the approach by adding the default values to the classes prop, when provided, it could reduce the noise and avoid an opportunity to forget or have the duplication done right.

```suggestion
      classes.root,
```

Reply:
The reason why I added the utility classes here is, if we want to use only the base (unstyled) slider we want this classes to be applied, we don't want this to depend on the wrapper component providing them. That way If I use only the unstyled version, I can target this classes for styling